python_scripts/linear_models_regularization.py

```diff
@@ -270,6 +272,55 @@
 # an effect on the performance. We chose the parameter beforehand and fix it
 # for the analysis.
 #
+# ### With categorical features
+#
+# Most of the time, categorical features will be encoded with a `OneHotEncoder`
+# when using a linear model. In practice, scaling the encoded features by a
+# `OneHotEncoder` is generally omitted since the encoded features are already
+# on a similar scale.
+#
+# Another aspect to consider when using a `OneHotEncoder` with linear models is
+# the link to feature collinearity. Indeed, with this encoding, one of the
+# encoded feature can be computed given all other features. For instance, for a
+# feature with two categories (e.g. "married" vs. "not married"), the feature
+# of the first category (i.e. "married") is the inverse of the feature of the
+# second category (i.e. "not married").
+
+# %%
+import numpy as np
+from sklearn.preprocessing import OneHotEncoder
+
+categories = np.array(
+    ["married"] * 5 + ["not_married"] * 5, dtype=object
+)[:, np.newaxis]
+
+encoder = OneHotEncoder(sparse=False, dtype=int)
```

Comment:
It's a bad practice to pass platform specific dtypes to numpy.

```suggestion
encoder = OneHotEncoder(sparse=False, dtype=np.int32)
```